pkg/controller/nodeset/node/controller.go

```diff
@@ -158,12 +160,8 @@ func New(name string, kubeClient kubernetes.Interface, nodesetClient nodesetclie
 			queueOwner(node)
 		},
 		DeleteFunc: func(obj interface{}) {
-			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-			if err == nil {
-				c.queue.Add(key)
-			}
-			node, err := c.nodeLister.Get(key)
-			if err != nil {
+			node, ok := obj.(*corev1.Node)
```

Comment:
are you sure about this change? I believe `obj` is a tombstone, not the actual object.

Reply:
I had a look while running with the debugger. It is a node-object which has the owner-reference

Reply:
I would bet I have seen controllers with this special logic

Reply:
@mrIncompetent can you recheck this?

Reply:
I will!

Reply:
So i checked and found:
```
// DeletedFinalStateUnknown is placed into a DeltaFIFO in the case where
// an object was deleted but the watch deletion event was missed. In this
// case we don't know the final "resting" state of the object, so there's
// a chance the included `Obj` is stale.
type DeletedFinalStateUnknown struct {
	Key string
	Obj interface{}
}
```

So that's why people use `DeletionHandlingMetaNamespaceKeyFunc`.
But how can we handle this case?
If we really run into this issue, the indexer wont contain the node probably. So we cannot get the owner reference & cannot put the nodeset into the queue.

Reply:
how do other controller do this?

Reply:
From the k8s replication controller: pkg/controller/replication/replication_controller.go:341
```golang
// When a pod is deleted, enqueue the ReplicationController that manages the pod and update its expectations.
// obj could be an *v1.Pod, or a DeletionFinalStateUnknown marker item.
func (rm *ReplicationManager) deletePod(obj interface{}) {
	pod, ok := obj.(*v1.Pod)

	// When a delete is dropped, the relist will notice a pod in the store not
	// in the list, leading to the insertion of a tombstone object which contains
	// the deleted key/value. Note that this value might be stale. If the pod
	// changed labels the new ReplicationController will not be woken up till the periodic resync.
	if !ok {
		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
			return
		}
		pod, ok = tombstone.Obj.(*v1.Pod)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a pod %#v", obj))
			return
		}
	}
...
```